stablehlo/reference/Ops.cpp

```diff
@@ -1576,6 +1581,14 @@ Tensor dotGeneralOp(const Tensor &lhs, const Tensor &rhs,
   return result;
 }
 
+Tensor dynamicIotaOp(Axis iotaDimension, const Tensor &outputShape,
+                     ShapedType resultType) {
+  if (resultType.hasStaticShape()) return iotaOp(iotaDimension, resultType);
```

Comment:
Followup: Let's move this error check to the top of the function, check all operands and all results. Make a ticket for now.

Reply:
tracking at https://github.com/openxla/stablehlo/issues/2294